planning/behavior_path_planner_common/src/utils/utils.cpp

```diff
@@ -503,6 +503,19 @@ bool isEgoOutOfRoute(
     if (utils::isInLanelets(goal_pose, shoulder_lanes)) {
       return !lanelet::utils::query::getClosestLanelet(shoulder_lanes, goal_pose, &goal_lane);
     }
+    lanelet::LaneletMapPtr map = route_handler->getLaneletMapPtr();
+    lanelet::ConstLanelets all_lanelets = lanelet::utils::query::laneletLayer(map);
+    lanelet::ConstLanelets route_lanelets = route_handler->getRouteLanelets();
+    for (auto it = route_lanelets.rbegin(); it != route_lanelets.rend(); ++it) {
+      for (auto it2 = all_lanelets.rbegin(); it2 != all_lanelets.rend(); ++it2) {
+        if ((*it).id() == (*it2).id()) {
+          //          std::cout << "Route lanelet id: " << (*it).id() << std::endl;
+          // select the lanelet goal lanelet
+          goal_lane = *it2;
+          return false;
+        }
+      }
+    }
```

Comment:
I think we can keep the goal pose.
For the goal_lane, if the behavior planner doesn't work without this modification we can keep it. However, I would like to add new parameter `use_differential_map_loading`, and update goal lane only when this parameter is set true.

Reply:
Hey @mitsudome-r,

I added the parameter you mentioned starting from `planning.launch.xml` first. I used this parameter both within the planning module and when specifying the input lanelet topic. Should I also include this parameter in `autoware.launch.xml`? If I add it to `autoware.launch.xml`, I believe I can use the same parameter for both the map and planning.